lighthouse-core/gather/gatherers/accessibility.js

```diff
@@ -50,46 +50,49 @@ function runA11yChecks() {
       'svg-img-alt': {enabled: false},
       'audio-caption': {enabled: false},
     },
-    // @ts-expect-error
-  }).then(axeResults => {
-    // axe just scrolled the page, scroll back to the top of the page so that element positions
-    // are relative to the top of the page
-    document.documentElement.scrollTop = 0;
+  });
 
-    // @ts-expect-error
-    const augmentAxeNodes = result => {
-      // @ts-expect-error
-      result.nodes.forEach(node => {
-        // @ts-expect-error - getNodeDetails put into scope via stringification
-        node.node = getNodeDetails(node.element);
-        // avoid circular JSON concerns
-        node.element = node.any = node.all = node.none = undefined;
-      });
+  // axe just scrolled the page, scroll back to the top of the page so that element positions
+  // are relative to the top of the page
+  document.documentElement.scrollTop = 0;
 
-      // Ensure errors can be serialized over the protocol
-      if (result.error instanceof Error) {
-        result.error = {
-          name: result.error.name,
-          message: result.error.message,
-          stack: result.error.stack,
-          errorNode: result.error.errorNode,
-        };
-      }
-    };
+  /** @param {import('axe-core/axe').Result} result */
+  const augmentAxeNodes = result => {
+    result.nodes.forEach(node => {
+      // @ts-expect-error - getNodeDetails put into scope via stringification
+      node.node = getNodeDetails(node.element);
+      // @ts-expect-error - avoid circular JSON concerns
+      node.element = node.any = node.all = node.none = undefined;
+    });
 
-    // Augment the node objects with outerHTML snippet & custom path string
-    axeResults.violations.forEach(augmentAxeNodes);
-    axeResults.incomplete.forEach(augmentAxeNodes);
+    // Ensure errors can be serialized over the protocol
+    /** @type {(Error & {mess: string, errorNode: any}) | undefined} */
+    // @ts-expect-error not on axe types
+    const error = result.error;
+    if (error instanceof Error) {
```

Comment:
TODO: there is no smoke test for this (can be detected with errorMessage = `axe-core Error`)

Reply:
well a smoke test here means leveraging an axe bug that hasn't been fixed, so that's not reasonable to keep maintaining. I simply verified that the rule error code hasn't changed in axe 4.

Reply:
When you say you verified, do you mean you did so manually?

Reply:
I looked at the axe code we linked to in the artifacts comment about `Accessibility` and confirmed that code wasn't changed in newer versions of axe.